lxd/daemon.go

```diff
@@ -198,17 +200,26 @@ func (d *Daemon) checkTrustedClient(r *http.Request) error {
 			Action: r.Method,
 		}}
 
-		_, err := authChecker.Allow(ctx, ops...)
-		return err
+		info, err := authChecker.Allow(ctx, ops...)
+		if err != nil {
+			return err, ""
+		}
+
+		if info != nil && info.Identity != nil {
+			return nil, info.Identity.Id()
+		}
```

Comment:
Under which conditions would ```authChecker.Allow()``` return a ```nil``` error **and** a ```nil``` info object? Also, under which conditions is info.Identity equals to ```nil```? My concern is that we silently swallow unexpected conditions, instead of signalling to the user that there's something wrong with it's auth request, but I really have no idea about ```authChecker.Allow()```.